core/modules.rs

```diff
@@ -292,6 +292,46 @@ impl ModuleLoader for NoopModuleLoader {
   }
 }
 
+pub struct InternalModuleLoader;
```

Comment:
Looking at this I'm confused as to how load ordering works. Are we essentially loading everything in the fixed order (as per number prefixes on the files)?

In the limit, we should just use ESM load ordering - that can be something for a follow up though.

Reply:
For the purposes of https://github.com/denoland/deno/pull/17672 we reworked how this is handled - `include_js_files!` will provide a list of files that the crates includes. The actual loading can be done lazily starting with the last included file and the module loader will take over to decide in which order it requests the files.